servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -158,31 +159,34 @@ private static final class CachingKeyFactory implements AsyncCloseable {
         }
 
         UrlKey get(final HttpRequestMetaData metaData) throws MalformedURLException {
-            final String host = metaData.host();
-            if (host == null) {
-                throw new MalformedURLException(
-                        "Request-target does not contain target host address: " + metaData.requestTarget() +
-                                ", expected absolute-form URL");
-            }
-
-            final String scheme = metaData.scheme();
-            if (scheme == null) {
-                throw new MalformedURLException("Request-target does not contains scheme: " +
-                        metaData.requestTarget() + ", expected absolute-form URL");
-            }
-
+            final String requestTarget = metaData.requestTarget();
+            final String scheme = ensureUrlComponentNonNull(metaData.scheme(), "scheme", requestTarget);
+            final String host = ensureUrlComponentNonNull(metaData.host(), "host", requestTarget);
             final int parsedPort = metaData.port();
             final int port = parsedPort >= 0 ? parsedPort :
                     (HTTPS_SCHEME.equalsIgnoreCase(scheme) ? defaultHttpsPort : defaultHttpPort);
             setHostHeader(metaData);
-            metaData.requestTarget(absoluteToRelativeFormRequestTarget(metaData.requestTarget(), scheme, host));
+            metaData.requestTarget(absoluteToRelativeFormRequestTarget(requestTarget, scheme, host));
 
             final String key = scheme + ':' + host + ':' + port;
             final UrlKey urlKey = urlKeyCache.get(key);
             return urlKey != null ? urlKey : urlKeyCache.computeIfAbsent(key, ignore ->
                     new UrlKey(scheme, HostAndPort.of(host, port)));
         }
 
+        private static String ensureUrlComponentNonNull(@Nullable final String value,
+                                                        final String name,
+                                                        final String requestTarget) throws MalformedURLException {
+            if (value == null) {
+                // 8 characters should be enough to give users an idea of what's wrong with the passed URL without
```

Comment:
Since we are only checking for scheme and host, what about omitting the path completely as part of the error?